crates/sovereign-sdk/module-system/sov-modules-stf-blueprint/src/stf_blueprint.rs

```diff
@@ -129,48 +128,39 @@ where
                     };
 
                     tx_receipts.push(receipt);
-                    continue;
+
+                    return Err(StateTransitionError::HookError(e));
                 }
             };
             // Commit changes after pre_dispatch_tx_hook
-            sc_workspace = sc_workspace.checkpoint().to_revertable();
+            // sc_workspace = sc_workspace.checkpoint().to_revertable();
 
-            let tx_result = self.runtime.dispatch_call(msg, &mut sc_workspace, &ctx);
+            let tx_result = self.runtime.dispatch_call(msg, sc_workspace, &ctx);
 
-            let events = sc_workspace.take_events();
+            // let events = sc_workspace.take_events();
             let tx_effect = match tx_result {
                 Ok(_) => TxEffect::Successful,
-                Err(e) => {
-                    native_error!(
-                        "Tx 0x{} was reverted error: {}",
-                        hex::encode(raw_tx_hash),
-                        e
-                    );
-                    // The transaction causing invalid state transition is reverted
-                    // but we don't slash and we continue processing remaining transactions.
-                    sc_workspace = sc_workspace.revert().to_revertable();
-                    TxEffect::Reverted
-                }
+                Err(e) => return Err(StateTransitionError::ModuleCallError(e)),
             };
             native_debug!("Tx {} effect: {:?}", hex::encode(raw_tx_hash), tx_effect);
 
             let receipt = TransactionReceipt {
                 tx_hash: raw_tx_hash,
-                events,
+                events: vec![],
```

Comment:
maybe we can open an issue to remove events from receipt and other places (if there isn't one already)

Reply:
#1570